### PR TITLE
[Docs] Fixed and improved QCL example

### PR DESCRIPTION
**Pull request by madagra**
Fixed the output of the QCL example after introducing the 2-dimensional output in the quantum model.

### CODE REVIEW

File: docs/qml/qcl.md [hunk 1]
```diff
@@ -10,21 +10,16 @@ shows how to fit the $\sin(x)$ function in the $[-1, 1]$ domain.
 In the following, train and test data are defined.
 
 ```python exec="on" source="material-block" session="qcl" result="json"
```
Comment (nmheim) [inline docs/qml/qcl.md:12]:
```suggestion
```python exec="on" source="material-block" session="qcl"
```

Comment (nmheim) [reply docs/qml/qcl.md:12]: removes the empty output box of the code block

File: docs/qml/qcl.md [hunk 2]
```diff
@@ -87,25 +84,27 @@ using a mean-square error loss, Adam optimizer. Training is performend on the GP
...
 
     optimizer.zero_grad()
 
-    loss = mse_loss(model(input_values), y_train)
+    # given a `n_batch` number of input points and a `n_observables`
+    # number of input observables to measure, the QNN returns
+    # an output of the following shape: [n_batch x n_observables]
+    # given that there is only one observable, a squeeze is applied to get
+    # a 1-dimensional tensor
+    loss = mse_loss(model(values=x_train).squeeze(), y_train)
```
Comment (nmheim) [inline docs/qml/qcl.md:107]: do we want to add an `assert loss < 1e-3` at the end to make sure this will error if it does not converge in a future docs build?

**Review by nmheim — APPROVED** [review]
tiny comments, pre-approving this :)
